xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/client/serverinfo/ServerInfo.java

```diff
@@ -0,0 +1,267 @@
+   * is optionally present in any server response and may be omitted.
+   *
+   * @return An optionally-present {@link BigDecimal} representing the cluster load factor.
+   */
+  @JsonProperty("load_factor_cluster")
+  Optional<BigDecimal> loadFactorCluster();
+
+  /**
+   * The current multiplier to the transaction cost that a transaction must pay to get into the open ledger. Per
+   * xrpl.org docs, this field is optionally present in any server response and may be omitted.
+   *
+   * @return An optionally-present {@link BigDecimal} representing the open ledger load factor.
+   */
+  @JsonProperty("load_factor_fee_escalation")
+  Optional<BigDecimal> loadFactorFeeEscalation();
+
+  /**
+   * The current multiplier to the transaction cost that a transaction must pay to get into the queue, if the queue is
+   * full. Per xrpl.org docs, this field is optionally present in any server response and may be omitted.
+   *
+   * @return An optionally-present {@link BigDecimal} representing the queue load factor.
+   */
+  @JsonProperty("load_factor_fee_queue")
+  Optional<BigDecimal> loadFactorFeeQueue();
+
+  /**
+   * The load factor the server is enforcing, not including the open ledger cost. Per xrpl.org docs, this field is
+   * optionally present in any server response and may be omitted.
+   *
+   * @return An optionally-present {@link BigDecimal} representing the server load factor.
+   */
+  @JsonProperty("load_factor_server")
+  Optional<BigDecimal> loadFactorServer();
+
+  /**
+   * How many other rippled servers this one is currently connected to.
+   *
+   * @return An optionally-present {@link UnsignedInteger} representing the number of peers of this server.
+   */
+  Optional<UnsignedInteger> peers();
+
+  /**
+   * (Admin only) Public key used by this node to sign ledger validations. This validation key pair is derived from the
+   * {@code [validator_token]} or {@code [validation_seed]} config field.
+   *
+   * @return A {@link String} containing the validator's public key.
+   */
+  @JsonProperty("pubkey_validator")
+  Optional<String> publicKeyValidator();
+
+  /**
+   * Information about the most recent fully-validated ledger. If the most recent validated ledger is not available, the
+   * response omits this field and includes {@link #closedLedger()} instead. Per xrpl.org docs, this field is
+   * optionally present in any server response and may be omitted.
+   *
+   * @return An optionally-present {@link ServerInfoLedger} representing the latest validated ledger.
+   */
+  @JsonProperty("validated_ledger")
+  Optional<ServerInfoLedger> validatedLedger();
+
+  /**
+   * Minimum number of trusted validations required to validate a ledger version. Some circumstances may cause the
+   * server to require more validations.
+   *
+   * @return An {@link UnsignedInteger} representing the quorum.
+   */
+  @JsonProperty("validation_quorum")
+  UnsignedInteger validationQuorum();
```

Comment:
Yes sir. It is always returned by all 3 models. There are 2 things that might tell us if we want to make a field optional: One that it appears sporadically in responses and second (may be omitted) in docs. This field seems to have none.

Reply:
I reached out to Rome and found out that, only the fields that say may be omitted and admin only should be optional for a general user, rest should be part of the response. So having the mandatory up in ServerInfo seems fine to me. Though the docs for server_info(clio) are incorrect.

Reply:
Hmm...I think you misunderstood. The `ServerInfo` interface should not allow non-optional fields.

Reason being is that we don't know what's going to come in the future, and if a future sub-class doesn't supply this value, then we will have to break our API again. Thus, to avoid any potential of API breakage, everything in `ServerInfo` must be optional (even if it happens to be in all 3 current sub-classes).

Now, if you think this value should be non-optional, then we need to pull it down 3 times into each sub-class, which I'm fine with if you are.